src/components/restaurant/Restaurant.js

```diff
@@ -0,0 +1,96 @@
+// Libraries
+import React from 'react'
+
+// lodash
+import _isEmpty from 'lodash/isEmpty'
+
+// components
+import SubHeader from '../subHeader'
+import FoodDescription from '../foodDescription'
+import Loader from "../../commonComponents/loader";
+import ErrorHandler from "../../commonComponents/errorHandler";
+
+// constants
+import { INITIAL_STATE, DEFAULT_RESTAURANT_ID, GENERAL_ERROR_MESSAGE } from "./constants/restaurant.general";
+
+// helpers
+import fetchRestaurantData from "../../api/fetchRestaurantData";
+
+// readers
+import restaurantReader from './readers/restaurantReader';
+
+// css
+import './restaurant.css'
+
+class Restaurant extends React.Component{
+
+    state = INITIAL_STATE;
+
+    componentDidMount(){
+        this.getRestaurantData();
+    }
+
+    getRestaurantData(){
+        fetchRestaurantData(DEFAULT_RESTAURANT_ID)
+        .then(this.setRestaurantData)
+        .catch(this.setErrorMessage)
+        .finally(this.setLoaded)
+    }
+
+    setRestaurantData = (restaurantData) => {
+        const restaurantDetails = restaurantReader.restaurantDetails(restaurantData);
+        const foodList = restaurantReader.foodList(restaurantData);
+        this.setState({ 
+            restaurantDetails,
+            foodList
+         });
+    }
+
+    setErrorMessage = (err) =>{
+        const { message=GENERAL_ERROR_MESSAGE } = err;
+        
+        this.setState({ error : message})
+    }
+
+    setLoaded = () =>{
+        this.setState({ isLoading : false })
+    }
+
+    renderSubheader(){
```

Comment:
try to have some gap between () and {
e.g; renderSubHeader() {.
    if(......) {  ... }
 }